src/software/ai/hl/stp/play/offense/offense_play.cpp

```diff
@@ -36,10 +35,27 @@ void OffensePlay::updateTactics(const PlayUpdate &play_update)
     unsigned int num_enemy_robots =
         static_cast<int>(play_update.world.enemyTeam().numRobots());
 
-    // enemy team has at most half a full team, so we need at most half the number of
-    // defenders
-    if (num_enemy_robots <= 3)
+    if (play_update.world.gameState().isSetupState() &&
+        (play_update.world.gameState().isOurDirectFree() ||
+         play_update.world.gameState().isOurIndirectFree()))
+    {
+        LOG(INFO) << "FREE KICK" << std::endl;
+        // if we're in free kick on the enemy side, then we can commit more robots in
+        // forward positions
+        if (play_update.world.ball().position().x() > 0)
+        {
+            num_defenders = std::max(
+                0, 2 - 2 * static_cast<int>(
+                               play_update.world.ball().position().x() /
+                               (play_update.world.field().fieldLines().xMax() / 3.0)));
```

Comment:
Could you add a comment explaining the conditions where there are 0 or 2 defending robots? From my understanding, if you pass a third of the enemy half, then there will be 0 defenders. Because, if the cast outputs 0, then we will have 2 robots defending, if it output 1+, then we will have 0 robots defending. If thats right and is what youre looking for, I feel like it can be simplified into a if, else statement.